test/e2e/cronjob_test.go

```diff
@@ -56,7 +59,57 @@ func TestSourceCronJob(t *testing.T) {
 	jpSinkRefNameInSpec := "jsonpath={.spec.sink.ref.name}"
 	out, err := test.getResourceFieldsWithJSONPath("cronjobsource", "testcronjobsource2", jpSinkRefNameInSpec)
 	assert.NilError(t, err)
-	assert.Equal(t, out, "testsvc1")
+	//assert.Equal(t, out, "testsvc1")
+	assert.Check(t, util.ContainsAllIgnoreCase(out, "testsvc1"))
+
+	t.Log("verify event messages from cronjob source")
+	mymsg := "This is a message from cronjob."
+	test.serviceCreateEventDisplay(t, r, "displaysvc")
+	test.cronJobSourceCreate(t, r, "testcronjobsource3", "*/1 * * * *", mymsg, "svc:displaysvc")
+	results := test.kn.Run("source", "cronjob", "describe", "testcronjobsource3")
+	r.AssertNoError(results)
+	out, err = kubectl{test.kn.namespace}.Run("get", "pod", "-l", "sources.eventing.knative.dev/cronJobSource=testcronjobsource3", "-o", "yaml")
+	t.Log(out)
+	err = test.verifyEventDisplayLogs(t, "displaysvc", "This is a message from cronjob")
+	assert.NilError(t, err)
+}
+
+func (test *e2eTest) verifyEventDisplayLogs(t *testing.T, svcname string, message string) error {
+	var (
+		retries int
+		err     error
+		out     string
+	)
+
+	selectorStr := fmt.Sprintf("serving.knative.dev/service=%s", svcname)
+	for retries < 5 {
+		out, err = kubectl{test.kn.namespace}.Run("logs", "-l", selectorStr, "-c", "user-container")
+		if err != nil {
+			t.Logf("error happens at kubectl logs -l %s -c -n %s: %v", selectorStr, test.kn.namespace, err)
+		} else if err == nil && strings.Contains(out, message) {
+			break
```

Comment:
why not return immediately ? then you don't have to check for nrRetries after the loop.